src/support/cleanse.h

```diff
@@ -8,6 +8,8 @@
 
 #include <stdlib.h>
 
+// Securely erase a span of memory in a way guaranteed not to be
```

Comment:
While it's true that there is nothing in the standard that guarantees this, there are numerous compiler-specific attributes for this, as well as libraries that guarantee this (like right now, memory_cleanse is just a wrapper around OPENSSL_cleanse)

My goal with adding the comment is to explain its purpose to other maintainers of the codebase, as it hopefully sees more use.